src/item/enchantment/VanillaEnchantments.php

```diff
@@ -59,47 +60,177 @@ final class VanillaEnchantments{
+		self::register("EFFICIENCY", new Enchantment(KnownTranslationFactory::enchantment_digging(), Rarity::COMMON,
+			Flags::DIG,
+			Flags::SHEARS,
+			5, false,
+			fn($level) => 10 * ($level - 1) + 1,
+			fn($level, $minCost) => $minCost + 50
+		));
+		self::register("FORTUNE", new Enchantment(KnownTranslationFactory::enchantment_lootBonusDigger(), Rarity::RARE,
+			Flags::DIG,
+			Flags::NONE,
+			3, false,
+			fn($level) => 9 * ($level - 1) + 15,
+			fn($level, $minCost) => $minCost + 50
+		));
+		self::register("SILK_TOUCH", new Enchantment(KnownTranslationFactory::enchantment_untouching(), Rarity::MYTHIC,
+			Flags::DIG,
+			Flags::SHEARS,
+			1, false,
+			fn($level) => 15,
+			fn($level, $minCost) => $minCost + 50
+		));
 
-		self::register("POWER", new Enchantment(KnownTranslationFactory::enchantment_arrowDamage(), Rarity::COMMON, ItemFlags::BOW, ItemFlags::NONE, 5));
-		self::register("PUNCH", new Enchantment(KnownTranslationFactory::enchantment_arrowKnockback(), Rarity::RARE, ItemFlags::BOW, ItemFlags::NONE, 2));
-		self::register("FLAME", new Enchantment(KnownTranslationFactory::enchantment_arrowFire(), Rarity::RARE, ItemFlags::BOW, ItemFlags::NONE, 1));
-		self::register("INFINITY", new Enchantment(KnownTranslationFactory::enchantment_arrowInfinite(), Rarity::MYTHIC, ItemFlags::BOW, ItemFlags::NONE, 1));
+		self::register("UNBREAKING", new Enchantment(KnownTranslationFactory::enchantment_durability(), Rarity::UNCOMMON,
+			Flags::ARMOR | Flags::DIG | Flags::SWORD | Flags::FISHING_ROD | Flags::BOW | Flags::TRIDENT | Flags::CROSSBOW,
+			Flags::SHEARS | Flags::FLINT_AND_STEEL | Flags::SHIELD | Flags::SMTH_ON_STICK | Flags::FISHING_ROD | Flags::ELYTRA | Flags::BRUSH,
+			3, false,
+			fn($level) => 8 * ($level - 1) + 5,
+			fn($level, $minCost) => $minCost + 50
+		));
+		self::register("MENDING", new Enchantment(KnownTranslationFactory::enchantment_mending(), Rarity::RARE,
+			Flags::NONE,
+			Flags::BREAKABLE,
+			1, true,
+			fn($level) => 25,
+			fn($level, $minCost) => $minCost + 50
+		));
+		self::register("VANISHING", new Enchantment(KnownTranslationFactory::enchantment_curse_vanishing(), Rarity::MYTHIC,
+			Flags::NONE,
+			Flags::ALL,
+			1, true,
+			fn($level) => 25,
+			fn($level, $minCost) => $minCost + 25
+		));
 
-		self::register("MENDING", new Enchantment(KnownTranslationFactory::enchantment_mending(), Rarity::RARE, ItemFlags::NONE, ItemFlags::ALL, 1));
-
-		self::register("VANISHING", new Enchantment(KnownTranslationFactory::enchantment_curse_vanishing(), Rarity::MYTHIC, ItemFlags::NONE, ItemFlags::ALL, 1));
-
-		self::register("SWIFT_SNEAK", new Enchantment(KnownTranslationFactory::enchantment_swift_sneak(), Rarity::MYTHIC, ItemFlags::NONE, ItemFlags::LEGS, 3));
+		self::register("SWIFT_SNEAK", new Enchantment(KnownTranslationFactory::enchantment_swift_sneak(), Rarity::MYTHIC,
+			Flags::NONE,
+			Flags::LEGGINGS,
+			3, true,
+			fn($level) => 10 * $level,
+			fn($level, $minCost) => $minCost + 5
+		));
 	}
```

Comment:
Seems to me like the min/max cost functions all have the same behaviour.

minCost is always (level - 1) * some constant + a fixed amount
maxCost is always the minCost + a fixed amount

Reply:
Not always. For example, for Respiration. There not $level - 1, but $level. Also enchantments such as Flame, Infinity essentially have no formula at all, return a single value (because the maximum enchantment level is 1)

Reply:
What we're saying here is that it's possible to use utility functions to factor identical logics.